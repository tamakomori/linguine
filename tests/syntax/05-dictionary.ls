func main() {
    // Dictionary literal
    a = {
        "aaa": "bbb",
        ccc: "ddd"
    };
    print(a.aaa);
    print(a["ccc"]);

    // Dictionary assignment
    dict = {};
    dict["aaa"] = "value1";
    dict["bbb"] = 123;

    // Dictionary remove
    for (k, v in dict) {
    	print("k = " + k + ", v = " + v);
    }
    print("unset");
    unset(dict, "aaa");
    for (k, v in dict) {
    	print("k = " + k + ", v = " + v);
    }
}
