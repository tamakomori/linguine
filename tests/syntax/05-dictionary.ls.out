bbb
ddd
k = aaa, v = value1
k = bbb, v = 123
unset
k = bbb, v = 123
